src/Domain/UseCase/GetSetting/GetSetting.ts

```diff
@@ -0,0 +1,33 @@
+import { inject, injectable } from 'inversify'
+import { GetSettingDto } from './GetSettingDto'
+import { GetSettingResponse } from './GetSettingResponse'
+import { UseCaseInterface } from '../UseCaseInterface'
+import TYPES from '../../../Bootstrap/Types'
+import { SettingProjector } from '../../../Projection/SettingProjector'
+import { SettingRepositoryInterface } from '../../Setting/SettingRepositoryInterface'
+
+@injectable()
+export class GetSetting implements UseCaseInterface {
+  constructor (
+    @inject(TYPES.SettingRepository) private settingRepository: SettingRepositoryInterface,
+    @inject(TYPES.SettingProjector) private settingProjector: SettingProjector,
+  ) {}
+
+  async execute(dto: GetSettingDto): Promise<GetSettingResponse> {
+    const { userUuid, settingName } = dto
+    const setting = await this.settingRepository.findOneByNameAndUserUuid(settingName, userUuid)
+    
+    if (setting === undefined) return {
```

Comment:
same here - could you turn inlines ifs into full blocks? Makes it hard to focus if we're using both styles.

also wouldn't `if (!setting)` be enough in this case?

Reply:
`if (!setting)` would do the same thing here, yes. But I like to be strict with types when I can help it/remember/it's cheap. So I'll leave as is unless you insist. ;)

Reply:
We use `isNullOrUndefined` util in SNJS https://github.com/standardnotes/snjs/blob/master/packages/snjs/lib/utils.ts#L112 👀

Ends up being pretty useful. I've had more than one rare bugs appear from using `if (!object)` syntax.

Reply:
Yeah, you get defensive about these things in JS. This is TS however and types are clearly defined. Even you would go crazy and add `null` as another alternative to the return type of `findOneByNameAndUserUuid` the typechecker would catch that below.

Anyway best not to mix `null`s and `undefined`s in one codebase and agree to use one or the other always to signify the absence of a value. `undefined` is fine by me.

Reply:
@mobitar I replied hastily and misread what you wrote, so ignore the first paragraph of my reply above. The second one stands and it's a way to avoid the same issues `isNullOrUndefined` is made to avoid. The nullish coalescing operator (`??`) also comes to mind. And with it a shorter name for the function: `isNullish`. ;)

Reply:
> Anyway best not to mix nulls and undefineds in one codebase 

I'm not sure but I feel like different APIs return different values. So localStorage would return null, but some other API would return undefined. I had rare bugs here as well, which is why I started using the util function.

> isNullish

👌

Reply:
@mobitar yeah, localStorage does return `null` for missing entries and this is indeed inconsistent with JS built-ins which pretty reliably return `undefined` to signify absence. This is a real shame.

The not mixing I advise applies to code written by us of course. If we stick to either `null` or `undefined` then we can rely on that for known functions. `undefined` might be better of the two, because it would be consistent with JS internals, so a mostly reliable default (less potential bugs).

And I would generally advise using `x === null | undefined | false | whatever` rather than `!x`, because:

* the intent is clear -- you want compare to a specific value that you expect is gotten from somewhere
* there is no relying on implicit conversion and danger that _they_ will cause unintended consequences

With this rule you are required to know exactly what your API returns -- that's where types help.

`isNullish` or `x === null || x === undefined` are sensible for APIs that may return both or when you don't care/don't want to know.